src/test/shared/settingsConfiguration.test.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import { extensionSettingsPrefix } from '../../shared/constants'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+
+describe('DefaultSettingsConfiguration', () => {
+    // These tests use an actual extension setting, because vscode.WorkspaceConfiguration fails when
+    // you attempt to update one that isn't defined in package.json. We will restore the setting value
+    // at the end of the tests.
+    const SETTING_KEY = 'telemetry'
+    let originalSettingValue: any
+
+    let sut: DefaultSettingsConfiguration
+
+    before(async () => {
+        originalSettingValue = vscode.workspace.getConfiguration(extensionSettingsPrefix).get(SETTING_KEY)
+    })
+
+    beforeEach(async () => {
+        sut = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+    })
+
+    after(async () => {
+        await vscode.workspace
+            .getConfiguration(extensionSettingsPrefix)
+            .update(SETTING_KEY, originalSettingValue, vscode.ConfigurationTarget.Global)
+    })
+
+    const scenarios = [
+        { testValue: 1234, desc: 'number' },
+        { testValue: 0, desc: 'default number' },
+        { testValue: 'hello world', desc: 'string' },
+        { testValue: '', desc: 'default string' },
+        { testValue: true, desc: 'true' },
+        { testValue: false, desc: 'false' },
+        { testValue: [], desc: 'empty array' },
+        { testValue: [{ value: 'foo' }, { value: 'bar' }], desc: 'array' },
+        { testValue: {}, desc: 'empty object' },
+        { testValue: { value: 'foo' }, desc: 'object' },
+        // Note: we don't test undefined because retrieval returns the package.json configured default value, if there is one
```

Comment:
VS Code doesn't care that the type for `aws.telemetry` is set to `string | null`?